make: Do not attempt to use color if the terminal cannot support it

fetmos opened the pull request:
This trivial change avoids annoying control codes damaging the screen if the terminal does not support color.

cgundogan commented on makefiles/cflags.inc.mk:
```diff
@@ -29,9 +29,11 @@ endif
 # disabled by the user
 CC_NOCOLOR ?= 0
 ifeq ($(CC_NOCOLOR),0)
+ ifneq ($(TERM),dumb)
```
You could set `CC_NOCOLOR` to `1` if a dumb term is detected. You could probably do that at around line 30-31. No need to do another wrap around this block,

cgundogan commented on makefiles/color.inc.mk:
```diff
@@ -4,16 +4,18 @@ COLOR_PURPLE :=
-    SHELL=bash
-  else
-    COLOR_ECHO   := /bin/echo -e
+ifneq ($(TERM),dumb)
```
it probably makes sense to have a more "general" variable here and set that to 1 as soon as a dumb term is detected, much like above for the cflags.

cgundogan commented:
Could you change your commit message to something like
```
make: disable color for dumb term
```
?
We usually prefix the commit message with a meaningful description for the subsystem that is touched.

miri64 commented:
Please prefix the the PR title and your commit message with the module the commit is changing.

miri64 commented:
(In your case not a module, but `make: ` ;-))

fetmos commented:
Okay: I pushed a new version which I think addresses all of your concerns, and makes the change more generic.  Actually it seems that this was what the original author intended but there was a small error because the return status of tput was used instead of the stdout.

cgundogan commented:
@fetmos out of interest: what's your `$TERM` set to and what returns `tput colors` for you?

cgundogan commented on makefiles/color.inc.mk:
```diff
@@ -1,10 +1,26 @@
 
-ifeq (0,  $(shell tput colors 2>&1 > /dev/null; echo $$?))
+ifeq ($(CC_NOCOLOR),)
+  available_colors:=$(shell tput colors)
```
`tput` can still fail when the terminfo file is borked. This would lead to some error messages during the invokation of `Make`. Could you redirect stderr to `/dev/null` like before? Can be verified with: `TERM="borked" tcap colors`.

fetmos replied:
Personally when errors occur I prefer to see the reason, rather than have the error message suppressed.  However I've put in the redirect as you requested, and pushed  a new commit.

cgundogan replied:
@fetmos I can see your concerns about error visibility. But when `tput` fails because of completely different reasons and your shell is otherwise in a valid state, then it will get pretty confusing especially for non advanced RIOT users. Since this check is just deciding about a colored output, I guess we can go with this way. Thanks for the update!

cladmi commented:
Works with `TERM=at` make.
